component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,383 @@
+                                "available \"rule-group-1\" resources or flushing the events if the " +
+                                "timeout(time.in.milliseconds:2000) occurs.\n" +
+                                "To aggregate the results from 'rule' queries, we have used 'and(not isValid)' " +
+                                "attribute aggregator where it logically computes AND operation of not isValid " +
+                                "boolean attribute values and outputs the results as a boolean value.\n" +
+                                "\n" +
+                                "Input 1: [SweetProductDefectsDetector]\n" +
+                                "{  \n" +
+                                "   \"event\":{  \n" +
+                                "      \"productId\":\"Cake\",\n" +
+                                "      \"colorCode\":\"FF0000\",\n" +
+                                "      \"height\": 5,\n" +
+                                "      \"width\": 10\n" +
+                                "\n" +
+                                "   }\n" +
+                                "}\n" +
+                                "\n" +
+                                "Output 1:[SweetProductDefectAlert]\n" +
+                                "{  \n" +
+                                "   \"event\":{  \n" +
+                                "      \"productId\":\"Cake\",\n" +
+                                "      \"isDefected\":\"false\"\n" +
+                                "   }\n" +
+                                "}\n" +
+                                "\n" +
+                                "Input 2: [SweetProductDefectsDetector]\n" +
+                                "{  \n" +
+                                "   \"event\":{  \n" +
+                                "      \"productId\":\"Cake\",\n" +
+                                "      \"colorCode\":\"FF0000\",\n" +
+                                "      \"height\": 10,\n" +
+                                "      \"width\": 20\n" +
+                                "\n" +
+                                "   }\n" +
+                                "}\n" +
+                                "\n" +
+                                "Output 2:[SweetProductDefectAlert]\n" +
+                                "{  \n" +
+                                "   \"event\":{  \n" +
+                                "      \"productId\":\"Cake\",\n" +
+                                "      \"isDefected\":\"true\"\n" +
+                                "   }\n" +
+                                "}"
+                )
+        }
+)
+public class ResourceBatchWindowProcessor extends WindowProcessor implements SchedulingProcessor {
+    private ComplexEventChunk<StreamEvent> currentEventChunk = new ComplexEventChunk<StreamEvent>(false);
+    private ComplexEventChunk<StreamEvent> eventsToBeExpired;
```

Comment:
Initialize this always

Reply:
Fixes added: d6242876caf0f65c7fd023c25c7c5e50e9fd7b3c